eng/native/functions.cmake

```diff
@@ -289,6 +300,20 @@ function(strip_symbols targetName outputFilename skipStrip)
   endif(CLR_CMAKE_HOST_UNIX)
 endfunction()
 
+function(install_symbols targetName destination_path)
```

Comment:
It is weird that we always strip symbols for libraries and strip them conditionally for coreclr. Since you are touching this code, I would take this opportunity to make the stripping unconditional. It should be fine (debugging should not be affected, I believe neither lldb nor gdb cares whether the symbols are in the .so files or in separate files) and it would also fix #32957 on OSX where debugging a core dump from checked / debug build is currently broken due to missing symbols.

Reply:
Should we start by removing the top-level option: https://github.com/dotnet/runtime/blob/8db7154c0e73010206807921452198357ac347e6/eng/native/build-commons.sh#L190

Reply:
II would do it in 3 steps in this order:
* Make stripping the symbols unconditional in the cmake files (essentially just ignore the `-stripsymbols` option as part of this PR
* Remove passing in the `-stripsymbols` to the build scripts in the CI / build lab scripts in a follow up PR
* Remove the `-stripsymbols` option support from the `build-commons.sh`

The point is that we don't want to have intermediate period where we would not be stripping the symbols and we also cannot pass in the -stripsymbols from the CI / build lab scripts after we remove support for that option from the  `build-commons.sh`

Reply:
Part 1 is pushed f193079. I will make followup PRs for 2 and 3. Thanks!